### PR TITLE
Candy Machine costs are wrong

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem

Found `cndy3Z4yapfJBmL3ShUp5exZKqR3z33thTzeNMm2gRZ` has very large cost - much larger than block max limit. Causing its transaction not land at all. 

#### Summary of Changes
This a draft change to identify/reproduce the issue. 

Fixes #

### CODE REVIEW

**Comment by t-nelson** [comment]
Interesting. So we were taking the max at the wrong point? I thought we were doing it _instead of_ the EMA in the error case

File: core/src/cost_update_service.rs [hunk 1]
```diff
@@ -157,8 +157,12 @@ impl CostUpdateService {
     ) -> u64 {
         let mut update_count = 0_u64;
         for (program_id, program_timings) in &mut execute_timings.details.per_program_timings {
-            let current_estimated_program_cost =
-                cost_model.read().unwrap().find_instruction_cost(program_id);
+            let current_estimated_program_cost = {
+                match cost_model.read().unwrap().find_instruction_cost_raw(program_id) {
```
Comment (carllin) [inline core/src/cost_update_service.rs:161]: nit: Can we rename this to `find_average_instruction_cost`, and rename the current `find_instruction_cost` to `find_inflated_instruction_cost()`

File: core/src/cost_update_service.rs [hunk 2]
```diff
@@ -157,8 +157,12 @@ impl CostUpdateService {
...
+                    Some((_, ema, _)) => ema.ceil() as u64,
+                    None => 200_000u64,
```
Comment (carllin) [inline core/src/cost_update_service.rs:163]: can we use a constant here? I.e. the block compute limit

**Comment by carllin** [comment]
When re-introducing the EMA changes, can we also add a datapoint error: https://github.com/solana-labs/solana/blob/0025f177f01ef957e4478b1437ba97710def82b3/runtime/src/execute_cost_table.rs#L105-L116 if the calculated cost gets higher than expected again? Should make debugging a little easier if something else blows up the cost.

File: core/src/cost_update_service.rs (hunk 2 above)
Comment (t-nelson) [reply core/src/cost_update_service.rs:163]: I think it needs to be responsive to the tx-wide-compute-limit feature being enabled as well

**Comment by taozhu-chicago** [comment]
> So we were taking the max at the wrong point? I thought we were doing it _instead of_ the EMA in the error case

wdym?

**Pull request closed without merging**